src/test/java/net/minecraftforge/debug/world/BiomeModifierTest.java

```diff
@@ -89,45 +95,73 @@ public BiomeModifierTest()
     {
         // Example of how to datagen datapack registry objects.
-        // Datapack registry objects referred to by other datapack registry objects must be registered first.
         DataGenerator generator = event.getGenerator();
         final Path outputFolder = generator.getOutputFolder();
-        final RegistryOps<JsonElement> ops = RegistryOps.create(JsonOps.INSTANCE, RegistryAccess.BUILTIN.get());
-        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        final RegistryOps<JsonElement> ops = RegistryOps.create(JsonOps.INSTANCE, RegistryAccess.builtinCopy());
         final String directory = PackType.SERVER_DATA.getDirectory();
-        Registry<PlacedFeature> placedFeatures = ops.registry(Registry.PLACED_FEATURE_REGISTRY).get();
 
-        // prepare to datagenerate our placed feature
+        // Prepare to datagenerate our placed feature.
         final String featurePathString = String.join("/", directory, MODID, Registry.PLACED_FEATURE_REGISTRY.location().getPath(), BASALT_PILLARS + ".json");
         final Path featurePath = outputFolder.resolve(featurePathString);
-        final PlacedFeature feature = placedFeatures.get(BASALT_PILLARS_RL);
-
-        // prepare to datagenerate our biome modifier
+        // Both casts here are needed for eclipse to compile.
+        final ResourceKey<ConfiguredFeature<?,?>> configuredFeatureKey = (ResourceKey<ConfiguredFeature<?,?>>)(ResourceKey<? extends ConfiguredFeature<?,?>>)NetherFeatures.LARGE_BASALT_COLUMNS.unwrapKey().get();
+        // Make sure we're using the holder from the registryaccess/registryops, the static ones won't work.
+        final Holder<ConfiguredFeature<?,?>> configuredFeatureHolder = ops.registry(Registry.CONFIGURED_FEATURE_REGISTRY).get().getOrCreateHolderOrThrow(configuredFeatureKey); 
+        final PlacedFeature feature = new PlacedFeature(
+            configuredFeatureHolder,
+            List.of(CountOnEveryLayerPlacement.of(1), BiomeFilter.biome()));
+        
         final ResourceLocation biomeModifiersRegistryID = ForgeRegistries.Keys.BIOME_MODIFIERS.location();
-        final String biomeModifierPathString = String.join("/", directory, MODID, biomeModifiersRegistryID.getNamespace(), biomeModifiersRegistryID.getPath(), MODIFY_BADLANDS + ".json");
+        final String biomeModifiersNamespace = biomeModifiersRegistryID.getNamespace();
+        final String biomeModifiersPath = biomeModifiersRegistryID.getPath();
+        final HolderSet.Named<Biome> badlandsTag = new HolderSet.Named<>(ops.registry(Registry.BIOME_REGISTRY).get(), BiomeTags.IS_BADLANDS);
+        final HolderSet.Named<Biome> forestsTag = new HolderSet.Named<>(ops.registry(Registry.BIOME_REGISTRY).get(), BiomeTags.IS_FOREST);
+        
+        // Prepare to datagenerate our add-feature biome modifier.
+        final String addFeaturePathString = String.join("/", directory, MODID, biomeModifiersNamespace, biomeModifiersPath, ADD_BASALT + ".json");
```

Comment:
Rather than having these long path strings could they be shortened by factoring some of it out into an `outputFolder.resolve(commonSuperPath)` and then storing that in a variable maybe?